mleap-xgboost-runtime/src/test/scala/ml/combust/mleap/xgboost/runtime/XGBoostPredictorRegressionModelParitySpec.scala

```diff
@@ -0,0 +1,109 @@
+import ml.combust.mleap.tensor.SparseTensor
+import ml.combust.mleap.xgboost.runtime.testing.{BoosterUtils, BundleSerializationUtils, CachedDatasetUtils, ClassifierUtils, FloatingPointApproximations, RegressionUtils}
+import ml.dmlc.xgboost4j.scala.Booster
+import org.apache.spark.ml.linalg.Vectors
+import org.scalatest.FunSpec
+import XgbConverters._
+
+
+class XGBoostPredictorRegressionModelParitySpec extends FunSpec
+  with BoosterUtils
+  with CachedDatasetUtils
+  with ClassifierUtils
+  with BundleSerializationUtils
+  with FloatingPointApproximations
+  with RegressionUtils {
+
+
+  /**
+   * A Predictor only provides a prediction column for performance reasons
+   */
+  def predictionColumnEqualityTest(mleapFrame1: DefaultLeapFrame, mleapFrame2: DefaultLeapFrame) = {
+    val predictionFrame1 = mleapFrame1.select("prediction").get
+    val predictionFrame2 = mleapFrame2.select("prediction").get
+
+    val predictionIndex = predictionFrame1.schema.indexOf("prediction").get
+
+    predictionFrame1.dataset zip predictionFrame2.dataset foreach {
+      case (row1, row2) => {
+        assert(
+          almostEqual(
+            row1.getDouble(predictionIndex),
+            row2.getDouble(predictionIndex)
+          )
+        )
+      }
+    }
+  }
+
+  it("We can deserialize an xgboost object into a Predictor by changing the MLeapOp") {
+    val xgboost4jTransformer = trainRegressor
+
+    val mleapBundle = serializeModelToMleapBundle(xgboost4jTransformer)
+    val deserializedPredictor: XGBoostPredictorRegression = loadXGBoostPredictorFromBundle(mleapBundle)
+      .asInstanceOf[XGBoostPredictorRegression]
+
+    assert(deserializedPredictor.model.predictor.isInstanceOf[Predictor])
+  }
+
+  it("A pre-serialization XGBoost4j model has the same results of a deserialized Predictor model") {
+    val xgboost4jTransformer = trainRegressor
+
+    val mleapBundle = serializeModelToMleapBundle(xgboost4jTransformer)
+    val deserializedPredictor: Transformer = loadXGBoostPredictorFromBundle(mleapBundle)
+
+    val preSerializationXGBoost4jResult = xgboost4jTransformer.transform(leapFrameBinomial).get
+    val predictorModelResult = deserializedPredictor.transform(leapFrameBinomial).get
+
+    predictionColumnEqualityTest(preSerializationXGBoost4jResult, predictorModelResult)
+  }
+
+  it("A deserialized XGBoost4j has the same results of a deserialized Predictor"){
+    val xgboost4jTransformer = trainRegressor
+
+    val mleapBundle = serializeModelToMleapBundle(xgboost4jTransformer)
+
+    val deserializedXGBoost4jTransformer: Transformer = loadMleapTransformerFromBundle(mleapBundle)
+    val deserializedXGBoost4jResult = deserializedXGBoost4jTransformer.transform(leapFrameBinomial).get
+
+    val deserializedPredictorTransformer: Transformer = loadXGBoostPredictorFromBundle(mleapBundle)
+    val deserializedPredictorResult = deserializedPredictorTransformer.transform(leapFrameBinomial).get
```

Comment:
Maybe add extra sanity checks here to assert that one is an instance of `XGBoost4j` and the other is `XGBoostPredictor`

Reply:
I've added extra sanity checks here.
Thank you for the feedback!